### ISSUE TITLE
Unable to start jGenProg

### ISSUE THREAD

**Issue by jphgoodwin** [SpoonLabs/astor#153]
Hi, I've compiled the jGenProg code and the Math-issue-280 example as per your instructions in your getting started doc(certainly not the clearest guide I've ever used). However, when I actually try to run jGenProg using the command:
~~~~
sudo java -cp $(cat /tmp/astor-classpath.txt):target/classes fr.inria.main.evolution.AstorMain -mode jgenprog -srcjavafolder /src/java/ -srctestfolder /src/test/ -binjavafolder /target/classes/ -bintestfolder  /target/classes/ -location /home/jphg/Documents/phd-work/astor-repository/astor/examples/Math-issue-280/ -dependencies examples/Math-issue-280/lib
~~~~

I get the following exception:
~~~~
2019-04-18 14:30:39,672 INFO fr.inria.astor.core.setup.ProjectConfiguration - Version of the JVM used: 1.8.0_191
2019-04-18 14:30:39,673 INFO main - Java version of the JDK used to run tests: 1.8.0_191
2019-04-18 14:30:39,673 INFO main - The compliance of the JVM is:  8
2019-04-18 14:30:39,675 INFO main - command line arguments: [-mode  jgenprog  -srcjavafolder  /src/java/  -srctestfolder  /src/test/  -binjavafolder  /target/classes/  -bintestfolder  /target/classes/  -location  /home/jphg/Documents/phd-work/astor-repository/astor/examples/Math-issue-280/  -dependencies  examples/Math-issue-280/lib]
2019-04-18 14:30:39,676 INFO fr.inria.main.evolution.AstorMain - Running Astor on a JDK at /usr/lib/jvm/java-8-openjdk-amd64/jre
[INFO ] fr.inria.main.AbstractMain.determineSourceFolders(AbstractMain.java:849) - Source folders: [/home/jphg/Documents/phd-work/astor-repository/astor/examples/Math-issue-280/src/java]
[INFO ] fr.inria.main.AbstractMain.determineSourceFolders(AbstractMain.java:868) - Source Test folders: [/home/jphg/Documents/phd-work/astor-repository/astor/examples/Math-issue-280/src/test]
[INFO ] fr.inria.astor.core.setup.ProjectConfiguration.addLocationToClasspath(ProjectConfiguration.java:140) - Adding to classpath junit-4.4.jar
[INFO ] fr.inria.astor.core.faultlocalization.gzoltar.GZoltarFaultLocalization.calculateSuspicious(GZoltarFaultLocalization.java:76) - -Executing Gzoltar classpath: /home/jphg/Documents/phd-work/astor-repository/astor/./output_astor/AstorMain-Math-issue-280//bin//default from 0 classes with test cases
[INFO ] fr.inria.astor.core.faultlocalization.gzoltar.GZoltarFaultLocalization.searchSuspicious(GZoltarFaultLocalization.java:150) - Gzoltar fault localization: min susp value parameter: 0.1
[INFO ] fr.inria.astor.core.faultlocalization.gzoltar.GZoltarFaultLocalization.searchSuspicious(GZoltarFaultLocalization.java:170) - -gz-Adding classpath: [/home/jphg/Documents/phd-work/astor-repository/astor/examples/Math-issue-280/lib/junit-4.4.jar, /home/jphg/Documents/phd-work/astor-repository/astor/examples/Math-issue-280/lib/, /home/jphg/Documents/phd-work/astor-repository/astor/./output_astor/AstorMain-Math-issue-280//bin//default]
[INFO ] fr.inria.astor.core.faultlocalization.gzoltar.GZoltarFaultLocalization.searchSuspicious(GZoltarFaultLocalization.java:219) - Gzoltar Test Result Total:0, fails: 0, GZoltar suspicious 0, with positive susp 0
[INFO ] fr.inria.astor.core.faultlocalization.gzoltar.GZoltarFaultLocalization.searchSuspicious(GZoltarFaultLocalization.java:227) - nr test results 0
[INFO ] fr.inria.astor.core.faultlocalization.gzoltar.GZoltarFaultLocalization.searchSuspicious(GZoltarFaultLocalization.java:272) - Gzoltar found: 0 with susp > 0.1, we consider: 0
Exception in thread "main" java.lang.IllegalArgumentException: No suspicious gen for analyze
	at fr.inria.astor.core.faultlocalization.gzoltar.GZoltarFaultLocalization.calculateSuspicious(GZoltarFaultLocalization.java:94)
	at fr.inria.astor.core.faultlocalization.gzoltar.GZoltarFaultLocalization.searchSuspicious(GZoltarFaultLocalization.java:50)
	at fr.inria.astor.core.solutionsearch.AstorCoreEngine.calculateSuspicious(AstorCoreEngine.java:899)
	at fr.inria.main.evolution.AstorMain.createEngine(AstorMain.java:125)
	at fr.inria.main.evolution.AstorMain.run(AstorMain.java:179)
	at fr.inria.main.evolution.AstorMain.execute(AstorMain.java:236)
	at fr.inria.main.evolution.AstorMain.main(AstorMain.java:210)
~~~~

Something to point out is I did all the compilation in Java 8, as I'm running Ubuntu 18.10 for which openjdk-7-jdk is no longer available in the snap repository and I was hoping I could at least get things running without it.

Do you think this is the cause, or is there something else that could be causing the problem?

**Comment by martinezmatias** [comment]
Hi @jphgoodwin 
Have you compiled the example (i.e. the program to repair)? (See in the doc how to do it)

> certainly not the clearest guide I've ever used
It would be a pleasure if you can help us to improve it :) PR are welcome ;)

Regards
Matias

**Comment by jphgoodwin** [comment]
Hi Matias,
Yes, I followed your instructions for compiling the example and I got this result at the end:
~~~~
Results :

Tests in error: 
  testMath280(org.apache.commons.math.distribution.NormalDistributionTest)

Tests run: 1979, Failures: 0, Errors: 1, Skipped: 0

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  16.368 s
[INFO] Finished at: 2019-04-18T16:06:53+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.4.3:test (default-test) on project commons-math: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/jphg/Documents/phd-work/astor-repository/astor/examples/Math-issue-280/target/surefire-reports for the individual test results.
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
~~~~
which I took from your notes was an expected error?
If I leave off the "test" argument from the maven command then the build goes through fine.

Regarding your getting started guide. If I manage to get Astor running I'll submit a pull request ;)
Thanks
James

**Comment by martinezmatias** [comment]
Dear James @jphgoodwin 
I think I found the problem: the command line you wrote seems to have an error:
the argument   `-bintestfolder  /target/classes/`  must be `-bintestfolder  /target/test-classes/ ` 

So the command would be:

`java -cp $(cat /tmp/astor-classpath.txt):target/classes fr.inria.main.evolution.AstorMain -mode jgenprog -srcjavafolder /src/java/ -srctestfolder /src/test/  -binjavafolder /target/classes/ -bintestfolder  /target/test-classes/ -location /home/user/astor/examples/Math-issue-280/ -dependencies examples/Math-issue-280/lib`


> Regarding your getting started guide. If I manage to get Astor running I'll submit a pull request ;)

Perfect!

Regards
Matias

**Comment by jphgoodwin** [comment]
Hi Matias,

That solves the problem thank you. I had previously run it with "test-classes", but was getting a different error at the time so made that change in an effort to resolve it. I think the previous problem might have been due to me not rebuilding the dependencies when I came back to it the next day after my tmp/ directory had been emptied.

Thanks for your help, I'll have a look at some suggestions for your getting started docs.
James
